google/datalab/storage/_object.py

```diff
@@ -147,6 +156,14 @@ def delete(self):
         self._api.objects_delete(self._bucket, self._key)
       except Exception as e:
         raise e
+      if wait_for_deletion:
+        while True:
+          objects = Objects(self._bucket, prefix=self.key, delimiter='/',
+                            context=self._context)
+          if any(o.key == self.key for o in objects):
+            time.sleep(_POLLING_SLEEP)
+            continue
```

Comment:
Do we want to have a timeout for this or are we sure the API never returns except in the success case?